### PR TITLE
Initial version of census rm docker dev

### PR DESCRIPTION
**Pull request by hbrace**
This is the adapted version of rasrm docker dev for census rm. Some services have been removed and some replaced by stubs.
The pubsub service has been added to rm-services.yml, and a pubsub emulator has been added to dev.yml. When you run make up, a script sets up a topic and subscription and you'll see if this is successful in the command line output.
All the image locations have been changed to 'eu.gcr.io/census-ci/rm/...' to use the new census images being stored in GCR (I haven't changed the iacsvc yet though, this should be done when it starts being stored on GCR).

https://trello.com/c/Ii0GEaR5/507-census-version-of-ras-rm-docker-dev-8

### CODE REVIEW

**Comment by hbrace** [comment]
[Census version of ras-rm-docker-dev (8)](https://trello.com/c/Ii0GEaR5/507-census-version-of-ras-rm-docker-dev-8)

File: .env [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+# EQ
+EQ_URL=https://eq-test/session?token=
+
+# PUBSUB Emulator
+PUBSUB_EMULATOR_HOSTNAME=pubsub-emulator
+PUBSUB_EMULATOR_PORT=8538
+PUBSUB_EMULATOR_HOST=${PUBSUB_EMULATOR_HOSTNAME}:${PUBSUB_EMULATOR_PORT}
+SUBSCRIPTION_PROJECT_ID=project
+RECEIPT_TOPIC_PROJECT_ID=project
+PUBSUB_EMULATOR_HOST=${PUBSUB_EMULATOR_HOSTNAME}:${PUBSUB_EMULATOR_PORT}
```
Comment (liamtoozer) [inline .env]: Think `PUBSUB_EMULATOR_HOST` is a duplicate from earlier on here?

File: ras-services.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+version: '3'
```
Comment (davidmort) [inline ras-services.yml]: i think we should add these into the rm-service.yml (or maybe have a census-rm-services.yml), as "ras" isn't part of RM for census

File: README.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
+# ras-rm-docker-dev
```
Comment (davidmort) [inline README.md]:
```suggestion
# census-rm-docker-dev
```

File: setup_database.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import os
+
```
Comment (AdeHarrison) [inline setup_database.py]: extra line needed here

**Comment by AdamHawtin** [comment]
The sample svc stub is missing, needs to be added for the current sample load process to work.

File: ras-services.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  party-service:
+    container_name: party-service
+    image: eu.gcr.io/census-ci/rm/census-rm-partysvc-stub
+    ports:
+      - "${PARTY_PORT}:${PARTY_PORT}"
+    networks:
+      - censusrmdockerdev_default
+    restart: always
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:${PARTY_PORT}/info"]
```
Comment (AdamHawtin) [inline ras-services.yml]: This healthcheck will always fail as the partysvc stub image does not have curl installed.

File: dev.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  links:
+   - ons-postgres:postgres
+ rasrm-ops:
+  container_name: rasrm-ops
+  image: eu.gcr.io/census-ci/rm/census-rm-ops
+  ports:
+  - "8003:80"
+ pubsub-emulator:
+  container_name: pubsub-emulator
+  image: bigtruedata/gcloud-pubsub-emulator
```
[the remark this answers is not in the document]
Comment (hbrace) [reply dev.yml]: Seems like ONSDigital have forked it already:
https://github.com/ONSdigital/docker-gcloud-pubsub-emulator
Worth building this image and pushing to GCR?

[the remark this answers is not in the document]
Comment (jcox-dev) [reply dev.yml]: @hbrace that was me! 👍

[the remark this answers is not in the document]
Comment (hbrace) [reply dev.yml]: I've built and pushed the emulator image to eu.gcr.io/census-ci/gcloud-pubsub-emulator. I haven't done any optimising of the image size though.

File: rm-services.yml [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+    container_name: pubsub
+    image: eu.gcr.io/census-ci/rm/census-rm-pubsub
+    external_links:
+      - rabbitmq
+      - pubsub-emulator
+    environment:
+      - RABBIT_AMQP=amqp://guest:guest@${RABBIT_HOST}:${RABBIT_PORT}
+      - SUBSCRIPTION_PROJECT_ID=project
+      - RECEIPT_TOPIC_PROJECT_ID=project
+      - PUBSUB_EMULATOR_HOST=${PUBSUB_EMULATOR_HOSTNAME}:${PUBSUB_EMULATOR_PORT}
```
[the remark this answers is not in the document]
Comment (hbrace) [reply rm-services.yml]: I've changed this back to the two variables as it doesn't seem to work with just using ${PUBSUB_EMULATOR_HOST}.

File: README.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+1. Make changes to whichever repository.  In this example we'll suppose you're changing the collection-instrument repository.
+1. Stop the service with `docker-compose -f ras-services.yml stop collection-instrument`
+1. Delete the stopped container with `docker-compose -f ras-services.yml rm collection-instrument`
+1. Rebuild the image and tag it as the latest to 'trick' the build into thinking we already have the latest and don't need to pull down the image from dockerhub.
+    1. Python repo - `docker build . -t eu.gcr.io/census-catd-ci/rm/collection-instrument/collection-instrument:latest`
+    1. Java repo - `mvn clean install` will automatically rebuild the docker image
+1. Finally, start the service again with `docker-compose -f ras-services.yml up -d collection-instrument`
+
+### Running natively with local changes
```
Comment (jcox-dev) [inline README.md]: Is this section still necessary / possible?

File: .env [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+PARTY_PASSWORD=secret
+
+# Collection Instrument
+COLL_INST_HOST=collection-instrument
+COLL_INST_PORT=8002
+COLL_INST_USERNAME=admin
+COLL_INST_PASSWORD=secret
+
+# Respondent Home UI
+ACCOUNT_SERVICE_URL=http://localhost:9092
```
Comment (liamtoozer) [inline .env]: Think this can be removed?

File: .env [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+SAMPLE_PORT=8125
+
+# Security
+SECURITY_USER_NAME=admin
+SECURITY_USER_PASSWORD=secret
+JWT_SECRET=testsecret
+SERVICE_ID=test_service_id
+
+# EQ
+EQ_URL=https://eq-test/session?token=
```
Comment (liamtoozer) [inline .env]: Think this can be removed?

**Pull request merged**